js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -304,7 +304,43 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
             const minContextSlot = options?.minContextSlot;
             try {
                 return await this.transact(async (wallet) => {
-                    await Promise.all([
+                    function getTargetCommitment() {
+                        let targetCommitment: Finality;
+                        switch (connection.commitment) {
+                            case 'confirmed':
+                            case 'finalized':
+                            case 'processed':
+                                targetCommitment = connection.commitment;
+                                break;
+                            default:
+                                targetCommitment = 'finalized';
+                        }
+                        let targetPreflightCommitment: Finality;
+                        switch (options?.preflightCommitment) {
+                            case 'confirmed':
+                            case 'finalized':
+                            case 'processed':
+                                targetPreflightCommitment = options.preflightCommitment;
+                                break;
+                            case undefined:
+                                targetPreflightCommitment = targetCommitment;
+                            default:
+                                targetPreflightCommitment = 'finalized';
+                        }
+                        const preflightCommitmentScore =
+                            targetPreflightCommitment === 'finalized'
+                                ? 2
+                                : targetPreflightCommitment === 'confirmed'
+                                ? 1
+                                : 0;
+                        const targetCommitmentScore =
+                            targetCommitment === 'finalized' ? 2 : targetCommitment === 'confirmed' ? 1 : 0;
+                        return preflightCommitmentScore < targetCommitmentScore
+                            ? targetPreflightCommitment
+                            : targetCommitment;
+                    }
+                    const [capabilities, _1, _2] = await Promise.all([
+                        wallet.getCapabilities(),
```

Comment:
Presumably there's nothing that should prohibit a client from firing off `get_capabilities` and `reauthorize` at the same time, right @sdlaver?